src/main/java/org/prebid/server/currency/CurrencyConversionService.java

```diff
@@ -155,6 +162,30 @@ public BigDecimal convertCurrency(BigDecimal price, Map<String, Map<String, BigD
         return price.divide(conversionRate, DEFAULT_PRICE_PRECISION, RoundingMode.HALF_EVEN);
     }
 
+    /**
+     *  Get conversion rate from request currency rates if it is present
+     */
+    private static BigDecimal getConversionRateFromRequest(Map<String, Map<String, BigDecimal>> requestCurrencyRates,
+                                                           String adServerCurrency, String effectiveBidCurrency,
+                                                           BigDecimal conversionRate) {
+        if (requestCurrencyRates != null) {
+            conversionRate = getConversionRate(requestCurrencyRates, adServerCurrency, effectiveBidCurrency);
+        }
+        return conversionRate;
+    }
+
+    /**
+     * Get conversion rate from server
+     */
+    private static BigDecimal getConversionRateFromServer(Map<String, Map<String, BigDecimal>> requestCurrencyRates,
```

Comment:
requestCurrencyRates not required here bc it can use externalValues from parameter.